01_network_sysapi/multithread/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+add_library(Threadguard INTERFACE)
+
+target_include_directories(Threadguard 
+	INTERFACE 
+		"${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+target_sources(Threadguard
```

Comment:
You don't have to list header files as sources. Just
```
add_library(name INTERFACE)
target_include_directories(name INTERFACE directory)
```
That's all, header-only target is ready.